snarkos/ledger/server.rs

```diff
@@ -162,6 +162,16 @@ impl<N: Network> Server<N> {
             .and(with(ledger.clone()))
             .and_then(Self::records_unspent);
 
+        // GET /testnet3/ciphertext/unspent/{commitment}
+        let get_unspent_ciphertext = warp::get()
```

Comment:
I'm thinking this shouldn't be its own entity separated from record, perhaps we can make it more uniform since they are different views of the same entity iiuc. we should try to avoid adding a specific endpoint for each particular client needs and favor more flexible general purpose endpoints.

for example:
/testnet3/records to get all records
/testnet3/records?spent=true for spent
/testnet3/records?spent=false for unspent
/testnet3/records/{commitment} for a single record by commitment
/testnet3/records/{commitment}?spent=false if we want the server to do the filtering part and return not found if it's doesn't exist or if it exist but is spent
/testnet3/records/{commitment}?spent=false&decrypt=false if we want the cyphertext

alternatively we could consider making the single records endpoint return both the decrypted data and the cyphertext as an attribute of the response payload?

maybe the wording or the specific endpoints I'm proposing are not the exact one, but I think we need to try to find a solution in that general direction

Reply:
@jrchatruc @raychu86 maybe you have input on this